src/components/components/AddLayerPanel/AddLayerPanel.component.jsx

```diff
@@ -0,0 +1,378 @@
+import React, { useState, useEffect } from 'react';
+import styles from './AddLayerPanel.module.scss';
+import classNames from 'classnames';
+import { Button } from '../Button';
+import { Chevron24Down, Load24Icon, Plus20Circle } from '../../../icons';
+import { Dropdown } from '../Dropdown';
+import CardPlaceholder from '../../../../assets/card-placeholder.svg';
+import { cardsData } from './cardsData';
+import {
+  createSvgExtrudedEntity,
+  createMapbox,
+  createStreetmixStreet,
+  create3DTiles,
+  createCustomModel,
+  createPrimitiveGeometry
+} from './createLayerFunctions';
+import Events from '/src/lib/Events';
+
+const AddLayerPanel = ({ onClose, isAddLayerPanelOpen }) => {
+  const [selectedOption, setSelectedOption] = useState(null);
+  const [groupedMixins, setGroupedMixins] = useState([]);
+
+  useEffect(() => {
+    // call getGroupedMixinOptions once time for getting mixinGroups
+    const data = getGroupedMixinOptions();
+    setGroupedMixins(data);
+  }, []);
+
+  // get all mixin data divided into groups, from a-mixin DOM elements
+  const getGroupedMixinOptions = () => {
+    const mixinElements = document.querySelectorAll('a-mixin');
+    const groupedArray = [];
+    let categoryName, mixinId;
+
+    const groupedObject = {};
+    let ind = 0;
+    for (let mixinEl of Array.from(mixinElements)) {
+      categoryName = mixinEl.getAttribute('category');
+      if (!categoryName) continue;
+      mixinId = mixinEl.id;
+      if (!groupedObject[categoryName]) {
+        groupedObject[categoryName] = [];
+      }
+      groupedObject[categoryName].push({
+        // here could be data from dataCards JSON file
+        img: '',
+        icon: '',
+        mixinId: mixinId,
+        name: mixinId,
+        id: ind
+      });
+      ind += 1;
+    }
+
+    for (let categoryName of Object.keys(groupedObject)) {
```

Comment:
This code is fine. But I just want to mention that `Object.keys` is creating a new array in memory, here in react that doesn't matter much. You can also use `for (const categoryName in groupedObject) {`. The for..in loop iterate over the keys of an object. There is one caveat is that it include inherited properties from the prototype chain, so you often need to use hasOwnProperty to filter out these properties. But here you use the object just as a data container so that's fine to not do the check for hasOwnProperty.